vsphere/resource_vsphere_resource_pool_test.go

```diff
@@ -0,0 +1,727 @@
+					rp, err := testGetResourcePool(s, "resource_pool")
+					if err != nil {
+						return "", err
+					}
+					return fmt.Sprintf("/%s/host/%s/Resources/terraform-resource-pool-test-parent/%s",
+						os.Getenv("VSPHERE_DATACENTER"),
+						os.Getenv("VSPHERE_CLUSTER"),
+						rp.Name(),
+					), nil
+				},
+				Config: testAccResourceVSphereResourcePoolConfigNonDefault(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccResourceVSphereResourcePoolCheckExists(true),
+					testAccResourceVSphereResourcePoolCheckCPUReservation(10),
+					testAccResourceVSphereResourcePoolCheckCPUExpandable(false),
+					testAccResourceVSphereResourcePoolCheckCPULimit(20),
+					testAccResourceVSphereResourcePoolCheckCPUShareLevel("custom"),
+					testAccResourceVSphereResourcePoolCheckCPUShares(10),
+					testAccResourceVSphereResourcePoolCheckCPUReservation(10),
+					testAccResourceVSphereResourcePoolCheckCPUExpandable(false),
+					testAccResourceVSphereResourcePoolCheckCPULimit(20),
+					testAccResourceVSphereResourcePoolCheckMemoryShareLevel("custom"),
+					testAccResourceVSphereResourcePoolCheckMemoryShares(10),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSphereResourcePool_tags(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccResourceVSphereResourcePoolPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceVSphereResourcePoolConfigTags(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccResourceVSphereResourcePoolCheckExists(true),
+					testAccResourceVSphereResourcePoolCheckTags("terraform-test-tag"),
+				),
+			},
+		},
+	})
+}
+
+func testAccResourceVSphereResourcePoolPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_DATACENTER") == "" {
+		t.Skip("set VSPHERE_DATACENTER to run vsphere_resource_pool acceptance tests")
+	}
+	if os.Getenv("VSPHERE_CLUSTER") == "" {
+		t.Skip("set VSPHERE_CLUSTER to run vsphere_resource_pool acceptance tests")
+	}
+	if os.Getenv("VSPHERE_ESXI_HOST5") == "" {
+		t.Skip("set VSPHERE_ESXI_HOST5 to run vsphere_resource_pool acceptance tests")
+	}
+}
+
+func testAccResourceVSphereResourcePoolCheckExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, err := testGetResourcePool(s, "resource_pool")
+		if err != nil {
+			if viapi.IsManagedObjectNotFoundError(err) && expected == false {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return errors.New("expected resource pool to be missing")
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereResourcePoolHasParent(parent string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rp, err := testGetResourcePool(s, "resource_pool")
+		if err != nil {
+			return err
+		}
+		prp, err := testGetResourcePool(s, parent)
+		if err != nil {
+			return err
+		}
+		props, err := resourcepool.Properties(rp)
```

Comment:
I usually consolidate these into a single properties function to avoid the extra error check. See [here](https://github.com/terraform-providers/terraform-provider-vsphere/blob/master/vsphere/helper_test.go#L765).